cpp/src/core/sroptcryst.h

```diff
@@ -98,53 +104,54 @@ class srTOptCryst : public srTGenOptElem {
 		//m_kMilND = srwlCr.h2;
 		//m_lMilND = srwlCr.h3;
 		//OC180314: the Miller indices are removed after discussion with A. Suvorov (because these are only required for the m_dA, and it is used as input parameter)
-
 		//m_tc = srwlCr.tc;
-		m_thicum = srwlCr.tc*1e+06; //assuming srwlCr.tc in [m]
-		
-		//ARF22053019 Geometry
-		m_uc = srwlCr.uc; //OC05092016
-		if((m_uc < 1) || (m_uc > 4)) throw IMPROPER_OPTICAL_COMPONENT_CRYSTAL_USE_CASE; //ARF 08052019 Change m_uc>2 to 4 for adding Laue cases
-
+        
+		//Thickness units [microns] ?
+		m_thicum = srwlCr.tc * 1e+06; //assuming srwlCr.tc in [m] goes to [um]
+        
+		//Geometry
+		m_uc = (int) srwlCr.uc; //OC05092016
+		if((m_uc < 1) || (m_uc > 4)) throw IMPROPER_OPTICAL_COMPONENT_CRYSTAL_USE_CASE; //ARF 08052019 Change m_uc>2 to 4 for adding Laue cases.
+		// Input #7: Whether to calculate the transmitted beam as well as the diffracted 
+		// beam. itrans = 0 for NO, itrans = 1 for YES.
+		m_ug = 1; //Bragg case ARF22053019
+		if ((m_uc == 3) || (m_uc == 4)) m_ug = 2; //Laue case ARF22053019
+            
 		m_itrans = 0; //OC: make it input variable
-		if((m_uc == 2) || (m_uc == 4)) m_itrans = 1; //ARF05082019 add m_uc == 4
-
-		double alphrd = srwlCr.angAs; //asymmetry angle [rad]
-
+		if ((m_uc == 2) || (m_uc == 4))  m_itrans = 1; //OC05092016 //ARF05082019 add m_uc == 4
+        
+        
+		alphrd = srwlCr.angAs; //asymmetry angle [rad]       
+        
 		// From Input #5: reciprocal lattice vector coordinates 
-		//m_HXAi[0] = 0.;
-		//m_HXAi[1] = cos(alphrd) / m_dA;
-		//m_HXAi[2] = -sin(alphrd) / m_dA;
-		//OC02102019, following ARF
-		if((m_uc == 1) || (m_uc == 2))
-		{//Bragg
-			m_HXAi[0] = 0.;
-			m_HXAi[1] = cos(alphrd)/m_dA;
-			m_HXAi[2] = -sin(alphrd)/m_dA;
-		}
-		else if((m_uc == 3) || (m_uc == 4))
-		{//Laue
-			//IP14062019 change recipr. lat. vector for Laue
+		m_HXAi[0] = 0.;
+		m_HXAi[1] = cos(alphrd) / m_dA;
+		m_HXAi[2] = -sin(alphrd) / m_dA;
+		if (m_ug == 2)
+		{
 			m_HXAi[0] = 0.;
-			m_HXAi[1] = -sin(alphrd)/m_dA;
+			m_HXAi[1] = sin(alphrd)/m_dA;
 			m_HXAi[2] = -cos(alphrd)/m_dA;
 		}
```

Comment:
Implemented, but without using "m_ug" member variable, and using "else if((m_uc == 3) || (m_uc == 4))" instead, as this was envisaged originally.